server/src/test/java/org/elasticsearch/common/lucene/store/InputStreamIndexInputTests.java

```diff
@@ -267,17 +268,46 @@ public void testSkipBytes() throws Exception {
             skipBytesExpected
         );
 
-        IndexInput input = dir.openInput("test", IOContext.DEFAULT);
-        InputStreamIndexInput is = new InputStreamIndexInput(input, limit);
+        var countingInput = new CountingReadBytesIndexInput("test", dir.openInput("test", IOContext.DEFAULT));
+        InputStreamIndexInput is = new InputStreamIndexInput(countingInput, limit);
         is.readNBytes(initialReadBytes);
         assertThat(is.skip(skipBytes), equalTo((long) skipBytesExpected));
+        long expectedActualBytesRead = Math.min(Math.min(initialReadBytes, limit), bytes);
+        assertThat(countingInput.getBytesRead(), equalTo(expectedActualBytesRead));
```

Comment:
tiniest of nits: I might have been inclined to do the assertion after reading the tail of the file, just to guard against an implementation that merely deferred the read instead of actually seeking. But this is pretty paranoid of me, it's fine the way it is :)

Reply:
Done